### PR TITLE
Fix three winding transformer calculation

### LINKED ISSUE #63: Correct calculation logic of three winding transformer

### PR DESCRIPTION
**Pull request by ckittl**
Resolves #63

### CODE REVIEW

**Review by danielfeismann — COMMENTED** [review]
LGTM but maybe there should be two real reviews rather then one and me :)

File: src/main/scala/edu/ie3/simona/model/grid/Transformer3wModel.scala [hunk 1]
```diff
@@ -319,66 +323,56 @@ case object Transformer3wModel {
...
       case PowerFlowCaseA =>
         (
-          transformerRefSystem.rInPu(transformerType.getrScA),
-          transformerRefSystem.xInPu(transformerType.getxScA),
-          transformerRefSystem.gInPu(transformerType.getgM),
-          transformerRefSystem.gInPu(transformerType.getbM).multiply(-1)
+          transformerType.getrScA,
+          transformerType.getxScA,
+          transformerType.getgM,
+          transformerType.getbM.multiply(-1)
```
[the remark this answers is not in the document]
Comment (ckittl) [reply src/main/scala/edu/ie3/simona/model/grid/Transformer3wModel.scala]: Good catch! Thanks!

**Review by danielfeismann — DISMISSED** [review]
LGTM Thank you :)

File: src/main/scala/edu/ie3/simona/model/grid/Transformer3wModel.scala [hunk 2]
```diff
@@ -319,66 +323,56 @@ case object Transformer3wModel {
...
+          .doubleValue()
         (
-          transformerRefSystem.rInPu(transformerType.getrScB),
-          transformerRefSystem.xInPu(transformerType.getxScB),
-          Quantities.getQuantity(0d, PU),
-          Quantities.getQuantity(0d, PU)
+          transformerType.getrScB.divide(pow(nominalRatio, 2)),
+          transformerType.getxScB.divide(pow(nominalRatio, 2)),
+          Quantities.getQuantity(0d, StandardUnits.ADMITTANCE),
+          Quantities.getQuantity(0d, StandardUnits.ADMITTANCE)
```
Comment (danielfeismann) [inline src/main/scala/edu/ie3/simona/model/grid/Transformer3wModel.scala:348]: with [PSDM#573](https://github.com/ie3-institute/PowerSystemDataModel/pull/573) we introduced Resistance, Reactance in addition to Impedance, as well as Conductance and Susceptance for Admittance. Does it make sense to use it here as well?

File: src/main/scala/edu/ie3/simona/model/grid/Transformer3wModel.scala [hunk 3]
```diff
@@ -319,66 +323,56 @@ case object Transformer3wModel {
...
+          .doubleValue()
         (
-          transformerRefSystem.rInPu(transformerType.getrScC),
-          transformerRefSystem.xInPu(transformerType.getxScC),
-          Quantities.getQuantity(0d, PU),
-          Quantities.getQuantity(0d, PU)
+          transformerType.getrScC.divide(pow(nominalRatio, 2)),
+          transformerType.getxScC.divide(pow(nominalRatio, 2)),
+          Quantities.getQuantity(0d, StandardUnits.ADMITTANCE),
+          Quantities.getQuantity(0d, StandardUnits.ADMITTANCE)
```
Comment (danielfeismann) [inline src/main/scala/edu/ie3/simona/model/grid/Transformer3wModel.scala:362]: Same here

**Review by danielfeismann — COMMENTED** [review]
Looks good to me, just one minor thing

File: src/main/scala/edu/ie3/simona/agent/grid/GridResultsSupport.scala [hunk 4]
```diff
@@ -447,12 +446,16 @@ private[grid] trait GridResultsSupport {
...
   ): PartialTransformer3wResult = {
-    val (iMag, iAng) = calcPortCurrent(
-      trafo3w,
-      nodeStateData.voltage,
+    val (_, iComplexPu) = iIJComplexPu(
       internalNodeStateData.voltage,
-      iNominal
+      nodeStateData.voltage,
+      yij(trafo3w),
+      Complex.zero,
```
[the remark this answers is not in the document]
Comment (danielfeismann) [reply src/main/scala/edu/ie3/simona/agent/grid/GridResultsSupport.scala]: Maybe both ways lead to the right result? See figure 4.13 in Chris Diss, for port B and C zero is correct in my opinion. Is this class also called for port A? May this case is coverd by class y0 in ransformer3wModel. 
`  /** Calculate the phase-to-ground admittance of a given transformer model. As
    * only the partial model for [[PowerFlowCaseA]] does contain a
    * phase-to-ground admittance at the internal port, in all other cases the
    * complex zero is returned
    *
    * @param transformer3wModel
    *   instance of [[Transformer3wModel]] the phase-to-ground admittance should
    *   be calculated for
    * @param port
    *   Port of the transformer, the result is requested for
    * @return
    *   phase-to-ground admittance Y_0 of the transformer model in p.u.
    */
  def y0(
      transformer3wModel: Transformer3wModel,
      port: Transformer3wPort.Value
  ): Complex = {
    val amount = transformer3wModel.amount
    transformer3wModel.powerFlowCase match {
      case PowerFlowCaseA if port.equals(Transformer3wPort.INTERNAL) =>
        val gij = transformer3wModel.gij().getValue.doubleValue()
        val bij = transformer3wModel.bij().getValue.doubleValue()
        val gii = transformer3wModel.g0().getValue.doubleValue()
        val bii = transformer3wModel.b0().getValue.doubleValue()
        amount * ((1 - 1 / transformer3wModel.tapRatio) * Complex(
          gij,
          bij
        ) + Complex(
          gii,
          bii
        ))
      case _ => Complex.zero
    }
  }`
 Still, your suggestion is the more understandable way, I think.

**Pull request merged**